src/console_cmds.cpp

```diff
@@ -75,11 +75,26 @@ static IntervalTimer<TimerGameCalendar> _scheduled_monthly_timer = {{TimerGameCa
 /**
  * Change a string into its number representation. Supports decimal and hexadecimal numbers.
  * @param arg The string to be converted.
+ * @param base The base for parsing the number.
  * @return The number, or std::nullopt when it could not be parsed.
  */
-static std::optional<uint32_t> ParseInteger(std::string_view arg)
+static std::optional<uint32_t> ParseInteger(std::string_view arg, int base = 0)
```

Comment:
The only useage making sense in hex are `TileIndex` in `ConResetTile`, `ConScrollToTile`.
So IMO make the `base` default to 10, and set it to 0 in those two uses.